src/game/Unit.cpp

```diff
@@ -8593,18 +8594,50 @@ void Unit::CleanupsBeforeDelete()
 
 CharmInfo* Unit::InitCharmInfo(Unit* charm)
 {
-    if (!m_charmInfo)
-        m_charmInfo = new CharmInfo(charm);
+    if (m_charmInfo)
```

Comment:
redundant null check before delete, except if you have it there to prevent double deletes but this is usually bad design: http://stackoverflow.com/questions/615355/is-there-any-reason-to-check-for-a-null-pointer-before-deleting

> setting the pointer explicitly to null after deleting might actually make sense depending on the architecture of the code, precisely to prevent double-deletes. Since this only saves double-deletes on the same pointer though, it's highly questionable whether this is a good design. I wouldn't ever rely on it.